### PR TITLE
[new-lair-2] encoding types to help with serialization / secrecy

### PR DESCRIPTION
**Pull request by neonphog**
DEPENDS ON https://github.com/holochain/lair/pull/60

### CODE REVIEW

File: crates/lair_keystore_api/src/encoding_types.rs [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+}
+
+impl BinDataSized<32> {
+    /// Treat this bin data as an ed25519 public key,
+    /// and use it to verify a signature over a given message.
+    pub async fn verify_detached<M>(
+        &self,
+        signature: BinDataSized<64>,
+        message: M,
+    ) -> LairResult<bool>
```
Comment (maackle) [inline crates/lair_keystore_api/src/encoding_types.rs:153]: Not a big deal but seems you don't need the self

```suggestion
    pub async fn verify_detached<M>(
        signature: BinDataSized<64>,
        message: M,
    ) -> LairResult<bool>
```

**Review by maackle — APPROVED** [review]
Guess I've been pretty `self`-absorbed the last 2 PRs, ha, ha

File: crates/lair_keystore_api/src/encoding_types.rs (hunk 1 above)
Comment (neonphog) [reply crates/lair_keystore_api/src/encoding_types.rs:153]: not sure I follow... how do you know what the public key is without `self`?

**Review by freesig — APPROVED** [review]
Looks good to me

**Pull request merged**